dev-python/python-augeas: bump to 1.0.3 and cleanup old

wimmuskee opened the pull request:
* 1.0.3 not working with python3_7
* existing ebuilds not working with setuptools (setup from distutils.core), but closing bug https://bugs.gentoo.org/678946 with the 1.0.3 addition (that one does require setuptools)
* dropping old because keyworded and newer 0.5.0 is partly stabilized

prometheanfire commented:
what makes you think py37 doesn't work with it?

prometheanfire commented:
to add to that, if you do change anything with it, I'd also change the cffi dep to 1.0.0 from 1.0.  If there IS a bug with py37 then you need do nothing.

Otherwise, lgtm.

wimmuskee commented:
I got a SandboxViolation error when testing with py37, and also travis upstream testing is not done with 3.7. I'll try testing with a py37 cffi, and i'll change the cffi version.

prometheanfire commented:
Sandbox violation seems odd, can you paste it?  normally that means a missing dep or something.

wimmuskee commented:
I think it probably has to do with not having the py37 target for cffi.

 * python3_7: running distutils-r1_run_phase distutils-r1_python_compile
python3.7 setup.py build -j 4
warning: no previously-included files found matching 'setup.pyc'
warning: no previously-included files matching 'yacctab.*' found under directory 'tests'
warning: no previously-included files matching 'lextab.*' found under directory 'tests'
warning: no previously-included files matching 'yacctab.*' found under directory 'examples'
warning: no previously-included files matching 'lextab.*' found under directory 'examples'
Traceback (most recent call last):
  File "/usr/lib/python3.7/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib/python3.7/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python3.7/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/usr/lib/python3.7/site-packages/setuptools/command/bdist_egg.py", line 172, in run
    cmd = self.call_command('install_lib', warn_dir=0)
  File "/usr/lib/python3.7/site-packages/setuptools/command/bdist_egg.py", line 158, in call_command
    self.run_command(cmdname)
  File "/usr/lib/python3.7/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/usr/lib/python3.7/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/usr/lib/python3.7/site-packages/setuptools/command/install_lib.py", line 11, in run
    self.build()
  File "/usr/lib/python3.7/distutils/command/install_lib.py", line 105, in build
    self.run_command('build_py')
  File "/usr/lib/python3.7/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/usr/lib/python3.7/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/usr/lib/python3.7/site-packages/setuptools/command/build_py.py", line 52, in run
    self.build_packages()
  File "/usr/lib/python3.7/distutils/command/build_py.py", line 372, in build_packages
    self.build_module(module, module_file, package)
  File "/usr/lib/python3.7/site-packages/setuptools/command/build_py.py", line 75, in build_module
    package)
  File "/usr/lib/python3.7/distutils/command/build_py.py", line 342, in build_module
    self.mkpath(dir)
  File "/usr/lib/python3.7/distutils/cmd.py", line 338, in mkpath
    dir_util.mkpath(name, mode, dry_run=self.dry_run)
  File "/usr/lib/python3.7/distutils/dir_util.py", line 70, in mkpath
    os.mkdir(head, mode)
  File "/usr/lib/python3.7/site-packages/setuptools/sandbox.py", line 310, in wrap
    path = self._remap_input(name, path, *args, **kw)
  File "/usr/lib/python3.7/site-packages/setuptools/sandbox.py", line 452, in _remap_input
    self._violation(operation, os.path.realpath(path), *args, **kw)
  File "/usr/lib/python3.7/site-packages/setuptools/sandbox.py", line 407, in _violation
    raise SandboxViolation(operation, args, kw)
setuptools.sandbox.SandboxViolation: SandboxViolation: mkdir('/var/tmp/portage/dev-python/python-augeas-1.0.3/work/python-augeas-1.0.3-python3_7/lib/pycparser', 511) {}

wimmuskee commented:
check, confirmed, except for some deprecation warnings, py37 testing running ok with cffi-1.11.5

have to get to work now, will update the PR later, thx for the feedback

prometheanfire commented:
yep, one other thing, update the EAPI to 7, other than that, good :D

prometheanfire commented on dev-python/python-augeas/python-augeas-1.0.3.ebuild:
```diff
@@ -0,0 +1,25 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
```
still needs to be eapi7

wimmuskee replied:
done

wimmuskee commented:
Added EAPI=7, and fixed the cffi dependency, also with a ${PYTHON_USEDEP}. It installs with py37 now. Removed the pypy3 support because cffi doesn't have it.
